gateway/src/main/scala/com/improving/app/gateway/domain/common/util.scala

```diff
@@ -39,100 +21,63 @@ object util {
     )
   }
 
-  def gatewayMemberInfoToMemberInfo(info: GatewayMemberInfo): MemberInfo = MemberInfo(
-    handle = info.handle,
-    avatarUrl = info.avatarUrl,
-    firstName = info.firstName,
-    lastName = info.lastName,
-    notificationPreference = Some(gatewayNotificationPreferenceToNotificationPreference(info.notificationPreference)),
-    notificationOptIn = info.notificationOptIn,
-    contact = Some(gatewayContactToContact(info.contact)),
-    organizationMembership = info.organizations.map(id => OrganizationId(id.toString)),
-    tenant = Some(TenantId(info.tenant.toString))
-  )
+  def gatewayMemberInfoToMemberInfo(info: GatewayMemberInfo): MemberInfo =
+    MemberInfo(
+      info.handle,
+      info.avatarUrl,
+      info.firstName,
+      info.lastName,
+      info.notificationPreference
+        .map(pref => NotificationPreference.fromValue(pref.value)),
+      info.notificationOptIn,
+      info.contact,
+      info.organizationMembership,
+      info.tenant
+    )
 
   def memberInfoToGatewayMemberInfo(info: MemberInfo): GatewayMemberInfo = GatewayMemberInfo(
-    handle = info.handle,
-    avatarUrl = info.avatarUrl,
-    firstName = info.firstName,
-    lastName = info.lastName,
-    notificationPreference = notificationPreferenceToGatewayNotificationPreference(info.notificationPreference),
-    notificationOptIn = info.notificationOptIn,
-    contact = contactToGatewayContact(info.contact.getOrElse(Contact.defaultInstance)),
-    organizations = info.organizationMembership.map(id => UUID.fromString(id.id)),
-    tenant = UUID.fromString(info.tenant.getOrElse(TenantId.defaultInstance).id)
-  )
-
-  private def gatewayContactToContact(contact: GatewayContact): Contact = Contact(
-    firstName = contact.firstName,
-    lastName = contact.lastName,
-    emailAddress = contact.emailAddress,
-    phone = contact.phone,
-    userName = contact.userName
+    info.handle,
+    info.avatarUrl,
+    info.firstName,
+    info.lastName,
+    info.notificationPreference.map(notificationPreferenceToGatewayNotificationPreference),
+    info.notificationOptIn,
+    info.contact,
+    info.organizationMembership,
+    info.tenant
   )
 
-  private def contactToGatewayContact(contact: Contact): GatewayContact = GatewayContact(
-    firstName = contact.firstName,
-    lastName = contact.lastName,
-    emailAddress = contact.emailAddress,
-    phone = contact.phone,
-    userName = contact.userName
-  )
-
-  private def memberStateToGatewayMemberState(memberStatus: MemberState): GatewayMemberStatus = {
-    if (memberStatus.isMemberStatusDraft) DRAFT_MEMBER_STATUS
-    else if (memberStatus.isMemberStatusActive) ACTIVE_MEMBER_STATUS
-    else if (memberStatus.isMemberStatusSuspended) SUSPENDED_MEMBER_STATUS
-    else TERMINATED_MEMBER_STATUS
+  private def memberStateToGatewayMemberState(memberStatus: MemberState): GatewayMemberStates = {
+    if (memberStatus.isMemberStatusDraft) GatewayMemberStates.MEMBER_STATES_DRAFT
```

Comment:
If you know a way to `match` while calling a function like `isMemberStatusDraft`, I am all ears. I tried setting this up as a match but IntelliJ recommended rewriting it this way.

Reply:
wow, the interface scalapb gives us for enums is not idiomatic at all. I wonder if a sealed oneof might be a better way to represent the state anyhow - that would allow us to include different data in the state as well as just the state type. I think the result is something a bit more idiomatic as well.

Reply:
I think this will be very useful because as it stands we cannot return a generic state for an endpoint. I ran into this earlier trying to view the Terminated state for Store - we will need to move all states to OneOfs instead of plain Scala, and use the derived states wherever we use the Scala states. That is, if we want to be able to return more than one kind of state from the same endpoint ;)